xbmc/filesystem/FavouritesDirectory.cpp

```diff
@@ -197,8 +197,15 @@ std::string CFavouritesDirectory::GetExecutePath(const CFileItem &item, const st
       execute = StringUtils::Format("ActivateWindow(%s,%s,return)", contextWindow.c_str(), StringUtils::Paramify(item.GetPath()).c_str());
   }
   //! @todo STRING_CLEANUP
-  else if (item.IsScript() && item.GetPath().size() > 9) // plugin://<foo>
+  else if (StringUtils::StartsWith(item.GetPath(), "script://"))
     execute = StringUtils::Format("RunScript(%s)", StringUtils::Paramify(item.GetPath().substr(9)).c_str());
+  else if (StringUtils::StartsWith(item.GetPath(), "plugin://"))
```

Comment:
Please use `item.IsPlugin()` and also make sure that it has a path after the protocol by checking the length.